### PR TITLE
removed `hasThisCache`

### PR DESCRIPTION
**Pull request by bekzod**
since `hasThisCache` only used `DEBUG` mode

### CODE REVIEW

File: packages/ember-metal/lib/property_get.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ export function get(obj, keyName) {
   assert(`Get must be called with two arguments; an object and a property key`, arguments.length === 2);
   assert(`Cannot call get with '${keyName}' on an undefined object.`, obj !== undefined && obj !== null);
   assert(`The key provided to get must be a string, you passed ${keyName}`, typeof keyName === 'string');
-  assert(`'this' in paths is not supported`, !hasThis(keyName));
+  assert(`'this' in paths is not supported`, keyName.lastIndexOf('this.', 0) !== 0);
```
Comment (stefanpenner) [inline packages/ember-metal/lib/property_get.js]: I'm not sure we have `lastIndexOf` support on all the browsers we support:http://caniuse.com/#search=lastIndexOf

In theory we could use a helper function to choose the built-in or fallback to polyfill?

Comment (stefanpenner) [inline packages/ember-metal/lib/property_get.js]: @krisselden would love your input on this change (From a performance stand point)

Comment (bekzod) [inline packages/ember-metal/lib/property_get.js]: `lastIndexOf` already have been used here https://github.com/emberjs/ember.js/pull/15390/files#diff-05b465f364bc76dfb813d27102cec7b6L9

Comment (stefanpenner) [inline packages/ember-metal/lib/property_get.js]: hmmm..... seems funky

Comment (rwjblue) [inline packages/ember-metal/lib/property_get.js]: Why?  `lastIndexOf` was ES5, we only support ES5 browsers...

Comment (stefanpenner) [inline packages/ember-metal/lib/property_get.js]: @rwjblue ah have we dropped pre IE11 support?

Comment (bekzod) [inline packages/ember-metal/lib/property_get.js]: @rwjblue It turns out `String.lastIndexOf` has been around since ES3 and `Array.lastIndexOf ` since ES5

Comment (rwjblue) [inline packages/ember-metal/lib/property_get.js]: @stefanpenner - I don't understand what you mean. IE9 was a fully ES5 browser...

Comment (rwjblue) [inline packages/ember-metal/lib/property_get.js]: @stefanpenner - In the link you shared, if you click "show all" you will see IE9 and IE10 listed as supported.

Comment (stefanpenner) [inline packages/ember-metal/lib/property_get.js]: @rwjblue silly me 📦

**Pull request merged**